python/lsst/meas/extensions/psfex/psfexPsfDeterminer.py

```diff
@@ -38,6 +38,44 @@
 from lsst.pipe.base import AlgorithmError
 
 
+class PsfexNoStarsError(AlgorithmError):
+    """Raised if no stars are available for PSF determination.
+
+    Parameters
+    ----------
+    msg : `str`
+        Error message.
+    """
+
+    def __init__(self, msg) -> None:
+        super().__init__(msg)
+
+    @property
+    def metadata(self) -> dict:
+        return {
+            "num_available_stars": 0,
```

Comment:
I don't think there's any metadata necessary for this exception: you can't do anything with this reported number, since it's always the same.